styleguide/STYLEGUIDE.md

```diff
@@ -1,50 +1,1090 @@
+
+  init(x: Int, y: Int, radius: Double) {
+    self.x = x
+    self.y = y
+    self.radius = radius
+  }
+
+  convenience init(x: Int, y: Int, diameter: Double) {
+    self.init(x: x, y: y, radius: diameter / 2)
+  }
+
+  override func area() -> Double {
+    return Double.pi * radius * radius
+  }
+}
+
+extension Circle: CustomStringConvertible {
+  var description: String {
+    return "center = \(centerString) area = \(area())"
+  }
+  private var centerString: String {
+    return "(\(x),\(y))"
+  }
+}
+```
+
+The example above demonstrates the following style guidelines:
+
+ + Specify types for properties, variables, constants, argument declarations and other statements with a space after the colon but not before, e.g. `x: Int`, and `Circle: Shape`.
+ + Define multiple variables and structures on a single line if they share a common purpose / context.
+ + Indent getter and setter definitions and property observers.
+ + Don't add modifiers such as `internal` when they're already the default. Similarly, don't repeat the access modifier when overriding a method.
+ + Organize extra functionality (e.g. printing) in extensions.
+ + Hide non-shared, implementation details such as `centerString` inside the extension using `private` access control.
+
+### Use of Self
+
+For conciseness, avoid using `self` since Swift does not require it to access an object's properties or invoke its methods.
+
+Use self only when required by the compiler (in `@escaping` closures, or in initializers to disambiguate properties from arguments). In other words, if it compiles without `self` then omit it.
+
+
+### Computed Properties
+
+For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided.
+
+**Preferred**:
+```swift
+var diameter: Double {
+  return radius * 2
+}
+```
+
+**Not Preferred**:
+```swift
+var diameter: Double {
+  get {
+    return radius * 2
+  }
+}
+```
+
+### Final
+
+Marking classes or members as `final` in tutorials can distract from the main topic and is not required. Nevertheless, use of `final` can sometimes clarify your intent and is worth the cost. In the below example, `Box` has a particular purpose and customization in a derived class is not intended. Marking it `final` makes that clear.
+
+```swift
+// Turn any generic type into a reference type using this Box class.
+final class Box<T> {
+  let value: T
+  init(_ value: T) {
+    self.value = value
+  }
+}
+```
+
+## Function Declarations
+
+Keep short function declarations on one line including the opening brace:
+
+```swift
+func reticulateSplines(spline: [Double]) -> Bool {
+  // reticulate code goes here
+}
+```
+
+For functions with long signatures, put each parameter on a new line and add an extra indent on subsequent lines:
+
+```swift
+func reticulateSplines(
+  spline: [Double], 
+  adjustmentFactor: Double,
+  translateConstant: Int, 
+  comment: String
+) -> Bool {
+  // reticulate code goes here
+}
```

Comment:
we're entering "very subjective" ground here, remember having exactly the same discussions in previous company ;) and I believe it's worth discussing within team scope. In such case personally I prefer using:

func reticulateSplines(spline: [Double],
                                      adjustmentFactor: Double ...